src/main/java/seedu/address/model/order/Order.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.model.order;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an Order made by a Person in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Order {
+
+    // Item fields
+    private final Item item;
+    private final Count count;
+
+    // Data fields
+    private final Date date;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Order(Item item, Count count, Date date) {
+        requireAllNonNull(item, count, date);
+        this.item = item;
+        this.count = count;
+        this.date = date;
+    }
+
+    public Item getDescription() {
```

Comment:
Maybe this method can be renamed to `getItem`? so we won't really get confused whether an order is supposed to have an item or a description.

Reply:
Thanks for this, change has been made in subsequent update.